spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/TelemetryTracker.java

```diff
@@ -42,11 +44,27 @@ public TelemetryTracker(Azure azure, String resourceGroup) {
         this.defaultProperties.put(PROPERTY_INSTALLATION_ID, TelemetryUtils.getHashMac());
     }
 
+    private TelemetryClient getTelemetryClient(TelemetryProperties telemetryProperties) {
+        final String instrumentationKey = telemetryProperties.getInstrumentationKey();
+
+        if (StringUtils.hasText(instrumentationKey)) {
+            final TelemetryClient client = new TelemetryClient();
+
+            client.getContext().setInstrumentationKey(instrumentationKey);
+
+            return client;
+        }
+
+        return null;
```

Comment:
Please don't return null directly, how about throw runtime exception?

Reply:
I think runtime exception may not suitable here, because telemetry logic should not effect on normal code.